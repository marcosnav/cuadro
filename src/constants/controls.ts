/**
 * Available control types
 */
export enum ControlType {
  NEW_GAME = 'NEW_GAME',
  RESTART_GAME = 'RESTART_GAME',
  SEE_ORIGINAL = 'SEE_ORIGINAL',
}
