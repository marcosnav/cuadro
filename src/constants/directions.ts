/**
 * SwipeDirection, one of UP, RIGHT, LEFT, DOWN
 */
export enum SwipeDirection {
  UP = 'UP',
  RIGHT = 'RIGHT',
  DOWN = 'DOWN',
  LEFT = 'LEFT',
}
