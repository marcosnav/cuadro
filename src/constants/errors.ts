export enum errors {
  NO_INTERNET = 'NO_INTERNET',
  UNEXPECTED_ERROR = 'UNEXPECTED_ERROR',
  UNRESPONSIVE_IMAGE_PROVIDER = 'UNRESPONSIVE_IMAGE_PROVIDER',
}
