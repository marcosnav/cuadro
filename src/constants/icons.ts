import check from './../assets/svg/check.svg';
import image from './../assets/svg/image.svg';
import newicon from './../assets/svg/new.svg';
import refresh from './../assets/svg/refresh.svg';

export const icons: { [key: string]: string } = {
  check,
  image,
  new: newicon,
  refresh,
};
