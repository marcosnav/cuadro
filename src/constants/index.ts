export * from './status';
export * from './errors';
export * from './theme';
export * from './keycodes';
