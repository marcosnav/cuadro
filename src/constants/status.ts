export enum status {
  LOADING_IMAGE = 'LOADING_IMAGE',
  DISPLAY_ORIGINAL = 'DISPLAY_ORIGINAL',
  PLAYING_GAME  = 'PLAYING_GAME ',
  FINISHED_GAME = 'FINISHED_GAME',
}
