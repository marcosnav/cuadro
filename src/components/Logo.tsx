import React, { FC } from 'react';
import { Status, theme } from './../constants';
import * as S from './styled';

interface IProps {
  status?: Status;
}

/**
 * Logo Component
 * @param props { status: Status }
 */
const Logo: FC<IProps> = ({ status }) => {
  const fill = status === Status.ERROR ? theme.RED : theme.BLUE;
  const animate = status === Status.LOADING_IMAGE || status === Status.STARTING_NEW_GAME;

  return (
    <S.LogoAnimator animate={animate}>
      <S.TriangleAnimator animate={animate}>
        <svg xmlns='http://www.w3.org/2000/svg' width='20' height='16' viewBox='0 0 20 16'>
          <path fill={fill} d='M10 0H0L10 16L20 0H10V2H16.5L10 12L3.5 2H10z'/>
        </svg>
      </S.TriangleAnimator>
      <S.TriangleAnimator reversed={true} animate={animate}>
        <svg xmlns='http://www.w3.org/2000/svg' width='20' height='16' viewBox='0 0 20 16'>
          <path fill={fill} d='M10 16H0L10 0L20 16H10V14H16.5L10 4L3.5 14H10z'/>
        </svg>
      </S.TriangleAnimator>
    </S.LogoAnimator>
  );
};

Logo.defaultProps = {
  status: Status.PLAYING_GAME,
};

export default Logo;
