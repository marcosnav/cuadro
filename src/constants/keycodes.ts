import { SwipeDirection } from './directions';

/**
 * KeyCodes to match for keypress events
 */
export enum KeyCodes {
  DOWN = 40,
  LEFT = 37,
  RIGHT = 39,
  UP = 38,
  ENTER = 13,
}

/**
 * Map key code with swipe direction
 */
export const KeySwipeMap: { [key: number]: SwipeDirection } = {
  40: SwipeDirection.DOWN,
  37: SwipeDirection.LEFT,
  39: SwipeDirection.RIGHT,
  38: SwipeDirection.UP,
};
