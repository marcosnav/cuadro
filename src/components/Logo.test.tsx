import { mount } from 'enzyme';
import 'jest-enzyme';
import 'jest-styled-components';
import React from 'react';
import { theme } from './../constants';
import Logo, { Status } from './Logo';

describe('<Logo/>', () => {
  test('renders properly', () => {
    const component = mount(<Logo />);
    expect(component).toMatchSnapshot();
  });

  describe('on: Status.LOADING', () => {
    test('animates for loading visual feedback', () => {
      const component = mount(<Logo status={Status.LOADING} />);
      expect(component).toHaveStyleRule('animation', expect.stringContaining('2s ease infinite'));
    });
  });

  describe('on: Status.ERROR', () => {
    test('figure fill must be apropriate', () => {
      const component = mount(<Logo status={Status.ERROR} />);
      const pathFills = component.find('path');
      expect(pathFills.length).toBe(2);
      expect(pathFills.getElements()[0].props.fill).toBe(theme.RED);
      expect(pathFills.getElements()[1].props.fill).toBe(theme.RED);
    });
  });
});
