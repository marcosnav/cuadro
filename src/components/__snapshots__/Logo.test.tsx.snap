// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo/> renders properly 1`] = `
.c0 {
  -webkit-animation: none;
  animation: none;
  height: 20px;
  position: relative;
  width: 20px;
}

.c2 {
  -webkit-animation: none;
  animation: none;
  left: 0;
  height: 16px;
  margin: auto;
  position: absolute;
  right: 0;
  top: 2px;
  width: 20px;
}

.c1 {
  -webkit-animation: none;
  animation: none;
  left: 0;
  height: 16px;
  margin: auto;
  position: absolute;
  right: 0;
  top: 2px;
  width: 20px;
  -webkit-animation: none;
  animation: none;
}

<Logo
  status="DEFAULT"
>
  <styled.div
    animate={false}
  >
    <StyledComponent
      animate={false}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  animation: ",
              [Function],
              "
  height: 20px;
  position: relative;
  width: 20px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        <Styled(styled.div)
          animate={false}
        >
          <StyledComponent
            animate={false}
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  animation: ",
                    [Function],
                    "
  left: 0;
  height: 16px;
  margin: auto;
  position: absolute;
  right: 0;
  top: 2px;
  width: 20px;
",
                    "
  animation: ",
                    [Function],
                    "
",
                  ],
                },
                "displayName": "Styled(styled.div)",
                "foldedComponentIds": Array [
                  "sc-bwzfXH",
                ],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="c1"
            >
              <svg
                height="16"
                viewBox="0 0 20 16"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 0H0L10 16L20 0H10V2H16.5L10 12L3.5 2H10z"
                  fill="#009CE5"
                />
              </svg>
            </div>
          </StyledComponent>
        </Styled(styled.div)>
        <styled.div
          animate={false}
        >
          <StyledComponent
            animate={false}
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "c2",
                  "rules": Array [
                    "
  animation: ",
                    [Function],
                    "
  left: 0;
  height: 16px;
  margin: auto;
  position: absolute;
  right: 0;
  top: 2px;
  width: 20px;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="c2"
            >
              <svg
                height="16"
                viewBox="0 0 20 16"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 16H0L10 0L20 16H10V14H16.5L10 4L3.5 14H10z"
                  fill="#009CE5"
                />
              </svg>
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</Logo>
`;
